File: client-web/src/api/system/fileSystem.ts
import { encodeURI } from '../../defs/tools'
import { API_BASE } from '../../config';

export async function checkFileSystem(path: string) {
  const res = await fetch(`${API_BASE}/system/check/filesystem/${encodeURI(path)}`);
  if (!res.ok) throw new Error('Failed to check filesystem');
  return res.json();
}

export async function listFiles(path: string, filter: string) {
  const res = await fetch(`${API_BASE}/system/list/files/${encodeURI(path)}/${encodeURI(filter)}`);
  if (!res.ok) throw new Error('Failed to list files');
  return res.json();
}

File: server/app/routers/system/fileSystem.py
from typing import Any
from app.config import conf_pathname

from fastapi import APIRouter, Depends
from sqlmodel import Session

from app.filesystem import check_file_exists, list_files
from app.database import get_session

from app.schemas import genericResponse

router = APIRouter()

@router.get(conf_pathname()+"/v1/system/check/filesystem/{path}", response_model=genericResponse)
def check_file(*, path: str) -> genericResponse:
    return check_file_exists(path = path)

@router.get(conf_pathname()+"/v1/system/list/files/{path}/{filter}", response_model=genericResponse)
def list_settings(*, path: str, filter: str | None = None) -> genericResponse:
    return list_files(path = path, filter = filter)

